### PR TITLE
refactor API endpoints

### LINKED ISSUE #31: Refactor `index.js`
`index.js` is starting to get pretty big (350 lines as of writing), so it might be worth doing a refactor pass (preferably after the outstanding PRs that edit this file are merged).

suggestion: create individual `.js` files in the `/api/` directory for each endpoint body.

### PR DESCRIPTION
**Pull request by CastroJ12**
please test it and let me know if anything is not working or if you have suggestions.

### CODE REVIEW

File: api/reservation.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+const express = require("express");
+const router = express.Router();
+
+// POST /api/item/reserve/:id endpoint
+router.post('/reserve/:id', async (req, res) => {
+
+    // Parse query and URL parameters
```
Comment (Bentheburrito) [inline api/reservation.js]: nitpick: It looks like the indentation is inconsistent here

File: api/return.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const express = require("express");
+const router = express.Router();
+
+// POST /api/item/return/:id endpoint
+router.post('/return/:id', async (req, res) => {
+    const id = parseInt(req.params.id);
```
Comment (Bentheburrito) [inline api/return.js]: also here: inconsistent indentation

File: api/login.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import jwt from 'jsonwebtoken';
```
Comment (diegovega49) [inline api/login.js:2]:
```suggestion
import bcrypt from 'bcrypt';
import jwt from 'jsonwebtoken';
```

File: api/register.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```
Comment (diegovega49) [inline api/register.js:3]:
```suggestion
import bcrypt from 'bcrypt';

const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```

File: index.js [hunk 5]
```diff
@@ -1,606 +1,62 @@
...
-  var { name, email, phone_number, password } = req.body;
+ // GET /api/item/:id endpoint
+ app.get('/api/item/:id', (req,res) => item_view(req,res,sql));
 
-  if (!email || !name || !password) {
-    res.status(400).json({ error: 'All_FIELDS_ARE_REQUIRED' });
-    return;
-  }
+ // POST /api/item/reserve/:id endpoint
+ app.post('/reserve/:id', (req,res) => reserveRoute(req,res,sql));
```
Comment (diegovega49) [inline index.js]:
```suggestion
 app.post('/api/item/reserve/:id', (req,res) => reserveRoute(req,res,sql));
```

File: index.js [hunk 6]
```diff
@@ -1,606 +1,62 @@
...
+import { confirmRoute } from './api/confirm-reservation.js';
+import { cancelRoute } from './api/cancel-reservation.js';
+import { returnRoute } from './api/return.js';
+import { listingRoute } from './api/listing.js';
+import { userRoute } from './api/user.js';
+import { loginRoute } from './api/login.js';
+import { registerRoute } from './api/register.js'
 
 const bodyParser = require('body-parser')
 const bcrypt = require('bcrypt')
```
Comment (diegovega49) [inline index.js]:
```suggestion
```

**Review by diegovega49 — CHANGES_REQUESTED** [review]

**Review by diegovega49 — APPROVED** [review]
Tested and LGTM

**Review by Bentheburrito — APPROVED** [review]
LGTM, tests pass locally

**Pull request merged**